src/directives/transfer-dom.js

```diff
@@ -1,22 +1,65 @@
 // Thanks to: https://github.com/airyland/vux/blob/v2/src/directives/transfer-dom/index.js
 // Thanks to: https://github.com/calebroseland/vue-dom-portal
 
+import { nanoid } from "nanoid";
+import Vue from "vue";
+
+function createMfePortalElement(guid) {
+    const mfeKey = Vue.prototype.$mfeKey;
+    if (!mfeKey) {
+        return null;
+    }
+
+    const portalId = getMfePortalId(guid);
+    const newPortal = document.createElement("div");
+    newPortal.id = portalId;
+    newPortal.classList = [mfeKey];
+    document.body.appendChild(newPortal);
+
+    return newPortal;
+}
+
+function getMfePortalElement(guid) {
+    const portalId = getMfePortalId(guid);
+    return document.getElementById(portalId)
+}
+
+function getMfePortalId(guid) {
+    return `mfe-portal-${guid}`;
+}
+
 /**
  * Get target DOM Node
  * @param {(Node|string|Boolean)} [node=document.body] DOM Node, CSS selector, or Boolean
  * @return {Node} The target that the el will be appended to
  */
-function getTarget (node) {
-    if (node === void 0) {
-        node = document.body
+function getTarget (node, guid) {
+    if (node !== void 0 && node !== true) {
+        return document.querySelector(node);
     }
-    if (node === true) { return document.body }
-    return node instanceof window.Node ? node : document.querySelector(node)
+
+    const mfeKey = Vue.prototype.$mfeKey;
+    if (mfeKey && guid) {
+        const portalElement = getMfePortalElement(guid);
+        if (portalElement) {
+            return portalElement;
+        }
+
+        return createMfePortalElement(guid);
```

Comment:
What was the reason for creating a new element if one doesn't already exist? Is having a side effect like this in a getter like this a problem?

Reply:
`getTarget` is just what the original method was called. The reason we need to create a new element if one doesn't already exist is because previously this `getTarget` method was returning the body element, which will always already exist. We are creating our own wrapper so if one doesn't exist we create one.